register.go

```diff
@@ -36,12 +36,18 @@ func (s *Snitch) register(looper director.Looper) error {
 		req.Audiences = append(req.Audiences, aud.ToProto())
 	}
 
-	var stream protos.Internal_RegisterClient
-	var err error
-	var quit bool
+	var (
+		stream             protos.Internal_RegisterClient
+		err                error
+		quit               bool
+		initialRegister    = true
+		initialRegisterErr error
+	)
 
+	// This might not error even if the handler returns an err - need to attempt
+	// to perform a recv to verify.
 	srv, err := s.serverClient.Register(s.config.ShutdownCtx, req)
-	if err != nil && !strings.Contains(err.Error(), context.Canceled.Error()) {
```

Comment:
Is there a reason why we shouldn't handle this as an err if context was cancelled?

Reply:
This was previously logspamming on service shutdown. Behavior might have changed now that reconnect logic is different